### PR TITLE
Improve velodrome balances logic

### PR DESCRIPTION
**Pull request by nicholasyoder**
Closes https://github.com/orgs/rotki/projects/11/views/3?pane=issue&itemId=143423154

improves a couple things:
* The `Failed to query velodrome locked balances for address` error was showing for accounts where there were velodrome deposits that didn't include the token ids (guage deposits and lock amount increases). Changed this so it won't attempt the query if there are no token ids found from the history for a given address.
* Changes the lock amount increase events to include the token ids as well. I observed on one account I was testing that the original lock event was from one address and the nft was then transfered to a different address where there were several amount increase events. Tracking only the second address would miss those token ids otherwise. A bit of an edge case probably but good to handle better.
* also adds to the docstring of the migration 22 test to mention why it doesn't include the api secret as discussed in [discord](https://discord.com/channels/657906918408585217/1080783409640648765/1449151567197503558)

### CODE REVIEW

File: rotkehlchen/chain/evm/decoding/velodrome/balances.py [hunk 1]
```diff
@@ -60,13 +60,15 @@ def get_gauge_address(self, event: 'EvmEvent') -> ChecksumEvmAddress | None:
 
     def query_balances(self) -> BalancesSheetType:
         balances = super().query_balances()
-        if len(addresses_with_deposits := self.addresses_with_deposits()) == 0:
+        if (
```
Comment (prettyirrelevant) [inline rotkehlchen/chain/evm/decoding/velodrome/balances.py:63]: why not create a dedicated `addresses_with_voting_escrow_deposits()` method to avoid iterating through events multiple times?

Comment (nicholasyoder) [reply rotkehlchen/chain/evm/decoding/velodrome/balances.py:63]: hmm... I guess in theory that would improve performance slightly, but I'm not sure its worth the bother here. Its not like its going to be looping over thousands of events (unless someone *really* uses velodrome a lot) so I doubt the performance improvement would be noticable most of the time. I think especially for a bugfixes pr like this its best to keep the logic fairly similar to what it was before to avoid breaking other unrelated things. If I'm not mistaken we loop over the events from addresses_with_deposits similarly in other balances logic as well. If its actually an issue perhaps we should have a note in develop for improving the way this is done in all balances logic, but as I said, not sure the performance improvement is enough to be worth the bother.
